src/main/scala/org/broadinstitute/dsde/firecloud/filematch/FileMatcher.scala

```diff
@@ -48,36 +58,41 @@ class FileMatcher extends LazyLogging {
     */
   private def performPairing(pathList: List[Path]): List[FileMatchResult] = {
     // find every path in the incoming pathList that is recognized by one of our known patterns
-    val desiredPairings: List[SuccessfulMatchResult] = findFirstFiles(pathList)
+    val matches = findFirstFiles(pathList)
+    val successfulMatches: List[SuccessfulMatchResult] = matches.collect { case x: SuccessfulMatchResult => x }
+    val partialMatches: List[PartialMatchResult] = matches.collect { case x: PartialMatchResult => x }
 
-    // remove the recognized firstFiles from the outstanding pathList
-    val remainingPaths: List[Path] = pathList diff desiredPairings.map(_.firstFile)
+    // remove the recognized firstFiles and partialMatches from the outstanding pathList
+    val remainingPaths: List[Path] =
+      pathList.diff(successfulMatches.map(_.firstFile) ++ partialMatches.map(_.firstFile))
 
     // process the recognized "read 1" files, and look for their desired pairings in the outstanding pathList.
     // this will result in either SuccessfulMatchResult when the desired pairing is found, or PartialMatchResult
     // when the desired pairing is not found
-    val pairingResults: List[FileMatchResult] = findSecondFiles(remainingPaths, desiredPairings)
+    val pairingResults: List[FileMatchResult] =
+      findSecondFiles(remainingPaths, successfulMatches.collect { case s: SuccessfulMatchResult => s })
```

Comment:
remove this `.collect` too, then all good!

Reply:
done: https://github.com/broadinstitute/firecloud-orchestration/pull/1581/commits/b2e49bb33fea29750828cc767dbcafe7cb9664e2